dashboard/src/actions/overviewActions.js

```diff
@@ -0,0 +1,200 @@
+import * as TYPES from "./types";
+import API from "../utils/axiosInstance";
+import { constructToast } from "./toastActions";
+export const getDatasets = () => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+
+    const username = await getState().userAuth.loginDetails.username;
```

Comment:
In some cases, username is not available so made it to await.

Reply:
I'm trying to understand...why would the username not be immediately available?  Is it because something has already requested it but it hasn't arrived yet (if so, what has requested it?), or is it because the user isn't logged in and we have to wait for the the user to finish interacting with the login page?  (How do we get to the overview page without having logged in?)

Reply:
In App.js, when the page gets loaded, an action `getUserDetails()` is dispatched to get the user details. If the `loginDetails` is not filled up and the `username` has been requested then it would be `undefined`. So made it to `await`

Reply:
Removed `await` to get the username

Reply:
I'm so glad that it worked without the `await`!  That's a huge relief!!  Thanks so much for trying it.